public/app/features/dashboard/partials/exportCsvModal.html

```diff
@@ -0,0 +1,32 @@
+<div class="modal-body" ng-controller="ExportCsvModalCtrl">
```

Comment:
We are not using this style of angular any more, moving to components (directives), with controllers that are bound in specific way (so no scope hierarchy) 

So do something like this:
this.publishAppEvent('show-modal', {
  templateHtml: '<data-export-modal data-list="dataList"></data-export-modal>',
  scope: { dataList: this.dataList  }
}); 

And then define a new data-export-modal component, use this for reference:
https://github.com/grafana/grafana/blob/master/public/app/core/components/help/help.ts
new to add the scope: {dataList: "="}  binding to the directive declaration